components/brave_rewards/common/rewards_flags.cc

```diff
@@ -83,18 +84,25 @@ RewardsFlags RewardsFlags::Parse(const std::string& input) {
   return flags;
 }
 
-const RewardsFlags& RewardsFlags::ForCurrentProcess() {
-  static RewardsFlags parsed_flags = [] {
-    std::string input;
+}  // namespace
 
-    const auto* command_line = base::CommandLine::ForCurrentProcess();
-    if (command_line->HasSwitch(kSwitchName)) {
-      input = command_line->GetSwitchValueASCII(kSwitchName);
-    }
+void RewardsFlags::ForceParsingForTesting() {
+  g_force_parse_for_testing = true;
+}
+
+RewardsFlags RewardsFlags::ForCurrentProcess() {
```

Comment:
How about:
```
template <bool testing = false>
static decltype(auto) ForCurrentProcess() {
  std::string input;

  const auto* command_line = base::CommandLine::ForCurrentProcess();
  if (command_line->HasSwitch(kSwitchName)) {
    input = command_line->GetSwitchValueASCII(kSwitchName);
  }

  if constexpr (testing) {
    return Parse(input);
  } else {
    static const RewardsFlags parsed_flags = Parse(input);
    return (parsed_flags);
  }
}
```
This way `decltype(RewardsFlags::ForCurrentProcess()) == const RewardsFlags&` and `decltype(RewardsFlags::ForCurrentProcess<true>()) == RewardsFlags`. So existing usages don't change, and you can call `RewardsFlags::ForCurrentProcess<true>()` in tests. You could also make the non-type template parameter an `enum class` if you want.

Reply:
@szilardszaloki, we do not call `RewardsFlags::ForCurrentProcess` from the tests so unfortunately this would not work.